lib/library.js

```diff
@@ -31,18 +31,20 @@ var EXT = Library.EXT = {
  * ForeignFunction.
  */
 
-function Library (libfile, funcs, lib) {
-  debug('creating Library object for', libfile)
+function Library (dl, funcs, lib) {
+  debug('creating Library object for', dl)
 
-  if (libfile && libfile.indexOf(EXT) === -1) {
-    debug('appending library extension to library name', EXT)
-    libfile += EXT
+  if (!(dl instanceof DynamicLibrary)) {
```

Comment:
IMO `instanceof` is an anti-pattern in dynamically typed languages like JavaScript. Consider a DynamicLibrary instance from another "env" (i.e. another copy of node-ffi in the node_modules structure), or another module that implements the DynamicLibrary interface but using a different backend (perhaps [libuv's dl functions](https://github.com/TooTallNate/node-dlopen)). Basically the "if it looks like a duck, and quacks like a duck" benefits of dynamic langs are lost with instanceof.